komorebi/src/windows_api.rs

```diff
@@ -300,14 +300,20 @@ impl WindowsApi {
                 }
             }
 
-            if monitors.elements().is_empty() {
-                monitors.elements_mut().push_back(m);
-            } else if let Some(preference) = index_preference {
-                while *preference > monitors.elements().len() {
+            if let Some(preference) = index_preference {
+                while *preference >= monitors.elements().len() {
                     monitors.elements_mut().push_back(Monitor::placeholder());
                 }
 
-                monitors.elements_mut().insert(*preference, m);
+                let current = monitors.elements_mut().get(*preference).expect(
```

Comment:
Let's do an if-let-some or a match here and log an error just in case - I've been bitten too many times by code paths I marked as unreachable before 😅

Reply:
Sure thing!